components/datetime/src/format/datetime.rs

```diff
@@ -65,20 +65,20 @@ pub struct FormattedDateTime<'l> {
 
 impl<'l> Writeable for FormattedDateTime<'l> {
     fn write_to<W: fmt::Write + ?Sized>(&self, sink: &mut W) -> fmt::Result {
-        write_pattern_plurals(
+        let _lossy = write_pattern_plurals(
             &self.patterns.get().0,
             self.date_symbols,
             self.time_symbols,
             &self.datetime,
             self.week_data,
             self.ordinal_rules,
-            self.fixed_decimal_format,
+            Some(self.fixed_decimal_format),
             sink,
         )
-        .map_err(|_e| {
+        .inspect_err(|_e| {
```

Comment:
Nit: Make sure to inspect the correct error, not the `core::fmt::Error`; put this after the `?`